crates/db_schema/src/impls/post.rs

```diff
@@ -302,34 +302,38 @@ impl Saveable for PostSaved {
   }
 }
 
-#[async_trait]
-impl Readable for PostRead {
-  type Form = PostReadForm;
-  async fn mark_as_read(
+impl PostRead {
+  pub async fn mark_as_read(
     pool: &mut DbPool<'_>,
-    post_read_form: &PostReadForm,
+    post_ids: Vec<PostId>,
```

Comment:
I like the idea of using a `HashSet` in these, to ensure uniqueness, but I leave that up to you.